.docker/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# start with official ruby docker image as base
+FROM ruby:3.1.2
+
+# set working directory within container
+WORKDIR /usr/src/app
+
+# pull in ruby (jekyll) and python (cite process) package info
+COPY Gemfile Gemfile.lock _cite/requirements.txt ./
+
+# install ruby packages
+RUN VERSION=$(grep -A 1 'BUNDLED WITH' Gemfile.lock | tail -n 1 | xargs); \
+    gem install bundler --version ${VERSION} && \
+    bundle _${VERSION}_ install
+
+# install python
+RUN apt update && apt install -y python3 python3-pip
+
+# install python packages
+RUN python3 -m pip install --upgrade --requirement requirements.txt
```

Comment:
Wouldn't this be be made irrelevant by virtue of Docker 1) caching the results of steps on re-runs anyway and 2) starting from scratch when the built-image is deleted completely and build anew? @falquaddoomi

Reply:
That's a good suggestion, actually. The benefit of adding `--no-cache-dir` is that the image ends up being slightly smaller due to pip not holding onto the archives. Usually the cache dir is used if you install a package in one project, then install it in another; in the second installation pip will pull it from the cache rather than having to redownload it. Since we're working within a single environment here, that won't ever occur, and you can save a little space by adding that flag.

Reply:
I've incorporated this on the upstream `tweaks` branch, and will incorporate on this pr in a second.